src/api/source/base-windshaft.js

```diff
@@ -42,17 +42,10 @@ export default class BaseWindshaft extends Base {
     _generateURL(auth, config) {
         let url = (config && config.serverURL) || DEFAULT_SERVER_URL_TEMPLATE;
         url = url.replace(/{user}/, auth.username);
-        this._validateServerURL(url.replace(/{local}/, ''));
+        validateServerURL(url.replace(/{local}/, ''));
         return {
             maps: url.replace(/{local}/, ':8181'),
```

Comment:
In my opinion, PORT numbers and any settings related should be in a config file. `/setup/config-service` could be a good place.